src/main/java/org/jabref/logic/formatter/Formatters.java

```diff
@@ -31,83 +31,63 @@
+    public static List<Formatter> getConverters() {
+        return Arrays.asList(
+                new HtmlToLatexFormatter(),
+                new HtmlToUnicodeFormatter(),
+                new LatexToUnicodeFormatter(),
+                new UnicodeToLatexFormatter()
+        );
     }
 
-    public static final List<Formatter> getCaseChangers() {
-        List<Formatter> caseChangers = new ArrayList<>();
-        caseChangers.addAll(CASE_CHANGERS);
-        return caseChangers;
+    public static List<Formatter> getCaseChangers() {
+        return Arrays.asList(
+                new CapitalizeFormatter(),
+                new LowerCaseFormatter(),
+                new SentenceCaseFormatter(),
+                new TitleCaseFormatter(),
+                new UpperCaseFormatter()
+        );
     }
 
-    public static final List<Formatter> getOthers() {
-        List<Formatter> others = new ArrayList<>();
-        others.addAll(OTHERS);
-        return others;
+    public static List<Formatter> getOthers() {
+        return Arrays.asList(
+                new ClearFormatter(),
+                new LatexCleanupFormatter(),
+                new MinifyNameListFormatter(),
+                new NormalizeDateFormatter(),
+                new NormalizeMonthFormatter(),
+                new NormalizeNamesFormatter(),
+                new NormalizePagesFormatter(),
+                new OrdinalsToSuperscriptFormatter(),
+                new RemoveBracesFormatter(),
+                new UnitsToLatexFormatter(),
+                new EscapeUnderscoresFormatter()
+        );
     }
 
-    public static final List<Formatter> getAll() {
+    public static List<Formatter> getAll() {
         List<Formatter> all = new ArrayList<>();
-        all.addAll(CONVERTERS);
-        all.addAll(CASE_CHANGERS);
-        all.addAll(OTHERS);
+        all.addAll(getConverters());
+        all.addAll(getCaseChangers());
+        all.addAll(getOthers());
         return all;
     }
 
     public static Optional<Formatter> getFormatterForModifier(String modifier) {
         Objects.requireNonNull(modifier);
-        Optional<Formatter> formatter;
-        List<Formatter> all = getAll();
 
-        if (modifier.matches("regex.*")) {
-            String regex = modifier.substring(LENGTH_OF_REGEX_PREFIX);
-            RegexFormatter.setRegex(regex);
-            formatter = all.stream().filter(f -> f.getKey().equals("regex")).findAny();
+        if (modifier.startsWith(RegexFormatter.KEY)) {
+            String regex = modifier.substring(RegexFormatter.KEY.length());
+            return Optional.of(new RegexFormatter(regex));
         } else {
-            formatter = all.stream().filter(f -> f.getKey().equals(modifier)).findAny();
-        }
-        if (formatter.isPresent()) {
-            return formatter;
+            Optional<Formatter> formatter = getAll().stream().filter(f -> f.getKey().equals(modifier)).findAny();
+            if (formatter.isPresent()) {
```

Comment:
That does not make any sense does it?
The method returns an optional. So you can directly return formatter.

Reply:
If no formatter is found, then we still want to try the switch at the end. I now reordered the code a bit and it should be clearer now. Good point.